fuseimg/data/ops/ops_mri.py

```diff
@@ -0,0 +1,981 @@
+
+def sort_dicoms_by_field(seq_path, dicom_field, use_order_indicator):
+    '''
+    Return location dir of requested sequence
+    '''
+
+    """
+     sort_dicom_by_dicom_field sorts the dcm_files based on dicom_field
+     For some MRI sequences different kinds of MRI series are mixed together (as in bWI) case
+     This function creates a dict={dicom_field_type:list of relevant dicoms},
+     than concats all to a list of the different series types
+     :param dcm_files: list of all dicoms , mixed
+     :param dicom_field: dicom field to sort based on
+     :return: sorted_names_list, list of sorted dicom series
+     """
+    dcm_files = glob.glob(os.path.join(seq_path, '*.dcm'))
+    dcm_values = {}
+    dcm_patient_z = {}
+    dcm_instance = {}
+    for index, dcm in enumerate(dcm_files):
+        dcm_ds = pydicom.dcmread(dcm)
+        patient_z = int(dcm_ds.ImagePositionPatient[2])
+        instance_num = int(dcm_ds.InstanceNumber)
+        try:
+            val = int(dcm_ds[dicom_field].value)
+            if val not in dcm_values:
+                dcm_values[val] = []
+                dcm_patient_z[val] = []
+                dcm_instance[val] = []
+            dcm_values[val].append(os.path.split(dcm)[-1])
+            dcm_patient_z[val].append(patient_z)
+            dcm_instance[val].append(instance_num)
+        except:
+            # sort by
+            if index == 0:
+                patient_z_ = []
+                for dcm_ in dcm_files:
+                    dcm_ds_ = pydicom.dcmread(dcm_)
+                    patient_z_.append(dcm_ds_.ImagePositionPatient[2])
+            val = int(np.floor((instance_num - 1) / len(np.unique(patient_z_))))
+            if val not in dcm_values:
+                dcm_values[val] = []
+                dcm_patient_z[val] = []
+                dcm_instance[val] = []
+            dcm_values[val].append(os.path.split(dcm)[-1])
+            dcm_patient_z[val].append(patient_z)
+            dcm_instance[val].append(instance_num)
+
+    sorted_keys = np.sort(list(dcm_values.keys()))
+    sorted_names_list = [dcm_values[key] for key in sorted_keys]
+    dcm_patient_z_list = [dcm_patient_z[key] for key in sorted_keys]
+    dcm_instance_list = [dcm_instance[key] for key in sorted_keys]
+
+    if use_order_indicator:
+        # sort from low patient z to high patient z
+        sorted_names_list_ = [list(np.array(list_of_names)[np.argsort(list_of_z)]) for list_of_names, list_of_z in
+                              zip(sorted_names_list, dcm_patient_z_list)]
+    else:
+        # sort by instance number
+        sorted_names_list_ = [list(np.array(list_of_names)[np.argsort(list_of_z)]) for list_of_names, list_of_z in
+                              zip(sorted_names_list, dcm_instance_list)]
+
+    return sorted_names_list_
+def _get_as_list(x):
+    if isinstance(x, list):
+        return x
+    return [x]
+
+############################
+class OpExtractLesionPropFromBBoxAnotation(OpBase):
```

Comment:
Is it crop operation?
Maybe it can just get as an input key to volume and key to bbox to crop.